pkg/component/etcd/crds/templates/crd-druid.gardener.cloud_etcds.yaml

```diff
@@ -1889,7 +1896,8 @@ spec:
     served: true
     storage: true
     subresources:
-      # /scale subresource is intentionally removed from this CRD, although it is specified
-      # in the original CRD from etcd-druid, due to adverse interaction with VPA.
-      # See https://github.com/gardener/gardener/pull/6850 for details.
-      status: {}
\ No newline at end of file
+      scale:
+        labelSelectorPath: .status.labelSelector
+        specReplicasPath: .spec.replicas
+        statusReplicasPath: .status.replicas
+      status: {}
```

Comment:
Back then the scale subresource was removed from the CRD to fix the following issue:
> This is a temporary fix for gardener, in order to allow vertical scaling of etcd pods once again.

See https://github.com/gardener/gardener/pull/6850.
cc @shreyas-s-rao @istvanballok to provider more details on what was the issue back then and why we still need to scale subresource to be disabled

Reply:
I'm not sure what to say. When the `scale` subresource does not work, why is it part of the official CRD definition?

Reply:
Scale subresource for etcd resource means that the VPA spec must target the etcd resource rather than the underlying statefulset like it currently does. Problem with this for the g/g usecase is that druid must only reconcile the etcd resource if the operation annotation is added to the etcd resource, or else druid would be reconciling the etcd resource all the time, causing possible downtimes to single-node etcd clusters.

So when VPA updates the etcd resources (memory and cpu requests), druid will not act upon it until the next maintenance window, which basically defeats the purpose of VPA for etcd in the first place. Hence, VPA targets the underlying statefulset instead. And due to the issue with VPA mentioned in https://github.com/gardener/gardener/pull/6850, we were forced to remove the scale subresource from the etcd CRD. We didn't want to remove it for the official CRD because there could be community users who use druid without the `gardener.cloud/operation` annotation requirement, so they can simply point their VPA to the etcd resource and druid would actively reconcile the etcd and update the underlying statefulset's resources accordingly.

Currently, @unmarshall and I are thinking of solutions to bypass this issue, so that VPA can point to the etcd resource itself, without triggering a full reconciliation of the etcd resource just for vertical scaling of the etcd pods. Until then, I would request that we keep the etcd scale subresource switched off.

Reply:
I feel the pain. Maybe you can drop the etcd CRDs from the PR for time being so that this PR does not regress the thing the colleagues mitigated back then.